src/lib/utils/orders.ts

```diff
@@ -0,0 +1,317 @@
+import dbConnect from "../dbConnect";
+import OrderModel from "../models/OrderModel";
+
+export const countOrders = async () => {
+  await dbConnect();
+  return await OrderModel.countDocuments();
+};
+
+export async function lastThirtyDaysOrdersPaid() {
+  await dbConnect();
+  const lastThirtyDaysOrdersPaid = await OrderModel.aggregate([
+    {
+      // Match paid orders within the last 30 days
+      $match: {
+        isPaid: true,
+        paidAt: {
+          $gte: new Date(new Date().setDate(new Date().getDate() - 30)),
```

Comment:
You can move this value to a const and reuse it in the following lines. Just to not repeat this line multiple times.